### ISSUE TITLE
More details of documentation intent

### ISSUE THREAD

**Issue by Pablion** [microsoft/vscode-docs#4906]
I see the current [code of intent](https://github.com/microsoft/vscode-docs/blob/04cec7670671ae852ef020991fb8441ee0bb2796/CONTRIBUTING.md#documentation-intent) like this: 

> The goal of the VS Code documentation is to educate users on VS Code features and how VS Code can be used to enhance their development experience with different programming languages and runtimes.
>
> The documentation is not intended to provide:
>
> - An introduction to coding or software development
> - Tutorials on technologies independent from VS Code
> - Promotion of third-party tools, plug-ins, or services
> - Excessive detail or advanced walkthroughs
>
> The documentation should target developers learning to use VS Code or searching for quick answers to commonly asked questions. Other forums such as blog posts can provide more detailed content elaborating on specific scenarios.

I wonder the `introduction to coding or software development` should not include how to develope a Visual Studio Code Extension (henceforth referred to as **"VSCE"**)

I made a Chrome Extension, and used AWS & Firebase before and I think we could learn something from their documentations. There are some problems I want to point out but I don't know if this is against the original intention:

1. #4905
2. Is the `"visibility": "collapsed"` a property of [contributes.views](https://code.visualstudio.com/api/references/contribution-points#contributes.views)? It is not mentioned 
3. Where to add the `when: "isLinux"` in [contributes.walkthroughs](https://code.visualstudio.com/api/references/contribution-points#contributes.walkthroughs)? I assume we can do it both under `walkthroughs` and under every `step`? Can we call it out? 
4. I want to see how the initial views look like right after the installation of the VSCE. I don't think F5 can do that. Nor `tsc -watch -p ./` does, and the VSCE being developed is not able to uninstall. I think there are some configuration file but I don't know where to remove/modify it.

In conclusion, I think the document can expalin more clearly if we are encouraged to make a more specific explain on each API.

**Comment by mjbvz** [comment]
TypeScript does not rewrite your import paths, that's left up to bundlers such as webpack / esbuild. The module options in the `tsconfig` exist mainly to help typescript understand how to resolve your imports

So ultimately you need to make sure that your extension is using imports that `node` can understand, either by writing these yourself to start with or by using a bundler that can generate them from your source

**Comment by Pablion** [comment]
Thank you @mjbvz. TBH, I'm kind of lost in the vscode internal debugger. I read [debugging](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations) and the [API page](https://code.visualstudio.com/api) but I don't know what kind of bundler is used here to write a vscode extension. 

I changed the `tsconfig.json` because I thought the bundler was reading it too. Obviously it's a misunderstanding.

**Comment by Pablion** [comment]
so what bundler do you use?

**Comment by gregvanl** [comment]
Hi @Pablion You can learn more about both esbuild and webpack in https://code.visualstudio.com/api/working-with-extensions/bundling-extension.
The `yo code` extension generator has the option to bundle a new extension with webpack.

**Comment by Pablion** [comment]
Thank you. I'll read it today. It feels like I can acquire any resources I need from the documentation now.. Maybe I was too stupid?

**Comment by gregvanl** [comment]
Hi @Pablion Keep up the good work! One of the great things about the VS Code APIs and documentation is that you can go as deep as you need to. One person can quickly make a small extension or color theme while someone else can create an extension with a full language service, a backend server, custom UI, and an elaborate testing and CI/CD pipeline. 
We hope people can find what they need, when they need it.
I'm going to close this issue but please reach out again if you find documentation issues or have suggestions.

**Comment by Pablion** [comment]
Yes I wanted to close this too since I don't know what I can offer. Although I do think there's some certain point to improve. Do you have any recommendation of books or learning materials on writing docs? My appreciation in advance.

**Comment by gregvanl** [comment]
One resource you might find helpful is the Microsoft Docs contributor guide at https://docs.microsoft.com/en-us/contribute/
It is written for people contributing to docs.microsoft.com but the writing recommendations apply for code.visualstudio.com.

**Comment by Pablion** [comment]
Thanks a lot!
